assets/dev/js/editor/elements/views/base.js

```diff
@@ -757,28 +757,42 @@ BaseElementView = BaseContainer.extend( {
 		let changed = false;
 
 		const renderDataBinding = async ( dataBinding ) => {
-			const { bindingSetting, bindingDynamicCssId } = dataBinding.dataset,
-				changedControl = this.getChangedDynamicControlKey( settings );
-			let change = settings.changed[ bindingSetting ];
+			const { bindingSetting, bindingConfig } = dataBinding.dataset;
+			// BindingSetting is kept for backward compatibility, should be removed once two versions from ED-17823 has been merged
+			// And use only the bindingConfig which contains all the needed information
+			const bindingSettings = bindingSetting.split( ' ' ); // Multiple binding settings can appear
+			const config = JSON.parse( bindingConfig );
+			let isEdited = false;
+
+			for ( const [ key, value ] of Object.entries( settings.changed ) ) {
```

Comment:
Maybe extract into a function called something like `getBindingChnagedValues` which will give you all the values that were changed and relevant to to the bindings (including dynamic). then loop over them and call `renderDataBoundChange`.
also it will make the let `isEdited` redundant.